mrmustard/lab_dev/states/base.py

```diff
@@ -36,6 +44,207 @@ class State(CircuitComponent):
+        r"""
+        Initializes a state from an ``(A, b, c)`` triple defining a Bargmann representation.
+
+        .. code-block::
+
+            >>> from mrmustard.physics.representations import Bargmann
+            >>> from mrmustard.physics.triples import coherent_state_Abc
+            >>> from mrmustard.lab_dev import Ket
+
+            >>> modes = [0, 1]
+            >>> triple = coherent_state_Abc(x=[0.1, 0.2])
+
+            >>> coh = Ket.from_bargmann(modes, triple)
+            >>> assert coh.modes == modes
+            >>> assert coh.representation == Bargmann(*triple)
+            >>> assert isinstance(coh, Ket)
+
+        Args:
+            modes: The modes of this states.
+            triple: The ``(A, b, c)`` triple.
+            name: The name of this state.
+            batched: Whether the given triple is batched.
+
+        Returns:
+            A state.
+
+        Raises:
+            ValueError: If the ``A`` or ``b`` have a shape that is inconsistent with
+                the number of modes.
+        """
+        raise NotImplementedError
+
+    @classmethod
+    def from_fock(
+        cls,
+        modes: Sequence[int],
+        array: ComplexTensor,
+        name: Optional[str] = None,
+        batched: bool = False,
+    ) -> State:
+        r"""
+        Initializes a state from an array describing the state in the Fock representation.
+
+        .. code-block::
+
+            >>> from mrmustard.physics.representations import Fock
+            >>> from mrmustard.physics.triples import coherent_state_Abc
+            >>> from mrmustard.lab_dev import Coherent, Ket
+
+            >>> modes = [0]
+            >>> array = Coherent(modes, x=0.1).to_fock_component().representation.array
+            >>> coh = Ket.from_fock(modes, array, batched=True)
+
+            >>> assert coh.modes == modes
+            >>> assert coh.representation == Fock(array)
+            >>> assert isinstance(coh, Ket)
+
+        Args:
+            modes: The modes of this states.
+            array: The Fock array.
+            name: The name of this state.
+            batched: Whether the given array is batched.
+
+        Returns:
+            A state.
+
+        Raises:
+            ValueError: If the given array has a shape that is inconsistent with the number of
+                modes.
+        """
+        raise NotImplementedError
+
+    @classmethod
+    def from_phasespace(
+        cls,
+        modes: Sequence[int],
+        cov: ComplexMatrix,
+        means: ComplexMatrix,
+        name: Optional[str] = None,
+        atol_purity: Optional[float] = 1e-3,
+    ) -> State:  # pylint: disable=abstract-method
+        r"""
+        Initializes a state from the covariance matrix and the vector of means of a state in
+        phase space.
+
+        Args:
+            cov: The covariance matrix.
+            means: The vector of means.
+            modes: The modes of this states.
+            name: The name of this state.
```

Comment:
```suggestion
        Initializes a ``State`` from the covariance matrix and the vector of means in phase space.
        This can take into account the ``s`` parameter of the representation: ``s=0`` (default) is Wigner W,
        ``s=1`` is Husimi Q, ``s=-1`` is Glauber P. Any other value is allowed, although unclear what it means.

        Args:
            cov: The covariance matrix in ``s`` representation.
            means: The vector of means in ``s`` representation.
            modes: The modes spanned by this state.
            s: The parameter of the representation: s=0 (default) is Wigner.
```

Reply:
We can add them in the future, but for the moment, the functions used for producing `Bargmann` (`wigner_to_bargmann_psi` and `wigner_to_bargmann_rho`) do not allow passing `s`. So I don't know what to do with `s` at the moment :(